docs/understand/architecture.md

```diff
@@ -57,6 +70,38 @@ The credential scheme, along with the request signing logic are compatible with
 
 Currently, the auth service manages its own database of users and credentials and does not use IAM in any way. 
 
-### Frontend UI
+### Hooks Engine
+
+The hooks engine enables CI/CD for data by triggering user defined [Actions](../setup/hooks.md) that will run during commit/merge. 
+
+### UI
 
 The UI layer is a simple browser-based client that uses the OpenAPI server. It allows management, exploration and data access to repositories, branches, commits and objects in the system.
+
+## Applications
+
+As a rule of thumb, lakeFS supports any s3-compatible application. This means that many common data applications work with lakeFS out of the box.
+
+## lakeFS Clients
+
+Some data applications require deeper integrations with lakeFS to support different use-cases or enhanced functionality which is provided by lakeFS clients.
+
+### OpenAPI Generated SDKs
+
+OpenAPI specification can be used to generate lakeFS clients for many programming languages.
+For example, the Python [lakefs-client](https://pypi.org/project/lakefs-client/) or the [Java client](https://search.maven.org/artifact/io.lakefs/api-client).
```

Comment:
Maybe state that we publish and maintain the Python, Java and Scala APIs and these are released on all new lakeFS releases.

Reply:
Done.